templates/federated_reporting/psql_wrapper.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# run an arbitrary psql command, either -f or -c
+# expectation is that a line will be output with exit_code=<n>
+# this exit code <n> will then be used to exit
+# This design enables cfengine policy to run a commands promise
+# that can be kept, repaired or failed depending on an exit
+# code from psql since psql can't get different codes from scripts.
+
+if [ $# -ne 2 ]
+then
+    echo "Usage: $0 [DBNAME] [SQL STRING]"
+    exit 2
+fi
+
+TMP=$(mktemp)
+TMP2=$(mktemp)
+OUT=$(psql --quiet --tuples-only --no-align --no-psqlrc "$1" -c "$2" 2> $TMP ; echo $? > $TMP2)
```

Comment:
Please double check that the `$?` is only expanded after the `psql`. I think it might be better to just assign it to the `RETURN_CODE` variable after the assignment to `OUT`.

Reply:
The exit code of the subshell will be the exit code of its last command.

Reply:
fixed